libs/server-sdk/src/data_interfaces/bootstrapper/ibootstrapper.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <launchdarkly/data_model/sdk_data_set.hpp>
+
+#include <tl/expected.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace launchdarkly::server_side::data_interfaces {
+
+/**
+ * Defines a component that can fetch a complete dataset for use in a
+ * Data System. Bootstrapping takes place when the SDK starts, and is
+ * responsible for provisioning the initial data that an SDK uses before it can
+ * begin the on-going synchronization process.
+ */
+class IBootstrapper {
+   public:
+    class Error {
+       public:
+        enum class Kind {
+            None,
+            Timeout,
+            Auth,
+        };
+
+        static Error Timeout(std::string detail) {
+            return Error(Kind::Timeout, std::move(detail));
+        }
+
+        static Error Auth(std::string detail) {
+            return Error(Kind::Auth, std::move(detail));
+        }
+
+        Error() : kind(Kind::None), detail(std::nullopt) {}
+
+       private:
+        Error(Kind kind, std::optional<std::string> detail)
+            : kind(kind), detail(std::move(detail)) {}
+        Kind kind;
+        std::optional<std::string> detail;
+    };
+
+    /**
+     * Fetch a complete dataset. This method must invokable multiple times.
```

Comment:
Can you clarify that second line?

Reply:
Realized that info is extraneous, will remove in subsequent PR.